pkg/security/admission/byrestrictions.go

```diff
@@ -0,0 +1,50 @@
+package admission
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// ByRestrictions is a helper to sort SCCs in order of most restrictive to least restrictive.
+type ByRestrictions []*kapi.SecurityContextConstraints
+
+func (s ByRestrictions) Len() int {
+	return len(s)
+}
+func (s ByRestrictions) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ByRestrictions) Less(i, j int) bool {
+	return s.pointValue(s[i]) < s.pointValue(s[j])
+}
+
+// pointValue places a value on the SCC based on the settings of the SCC that can be used
+// to determine how restrictive it is.  The lower the number, the more restrictive it is.
+func (s ByRestrictions) pointValue(constraint *kapi.SecurityContextConstraints) int {
+	points := 0
+
+	// make sure these are always valued higher than the combination of the highest strategies
+	if constraint.AllowPrivilegedContainer {
+		points += 10
```

Comment:
Agree since w/ priv'd container and root uid you can escape container and
mount whatever you want
On Fri, Jun 12, 2015 at 8:51 PM Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/security/admission/byrestrictions.go
> <https://github.com/openshift/origin/pull/2868#discussion_r32365757>:
>
> > +func (s ByRestrictions) Len() int {
> > +	return len(s)
> > +}
> > +func (s ByRestrictions) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
> > +func (s ByRestrictions) Less(i, j int) bool {
> > +	return s.pointValue(s[i]) < s.pointValue(s[j])
> > +}
> > +
> > +// pointValue places a value on the SCC based on the settings of the SCC that can be used
> > +// to determine how restrictive it is.  The lower the number, the more restrictive it is.
> > +func (s ByRestrictions) pointValue(constraint *kapi.SecurityContextConstraints) int {
> > +	points := 0
> > +
> > +	// make sure these are always valued higher than the combination of the highest strategies
> > +	if constraint.AllowPrivilegedContainer {
> > +		points += 10
>
> @pmorie <https://github.com/pmorie> opinion?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/2868/files#r32365757>.
>